### ISSUE TITLE
Launcher is taking too much memory

### ISSUE THREAD

**Issue by hmittal90** [amirzaidi/Launcher3#205]
Thanks for awesome launcher.

I observe that launcher is taking too much memory.can we optimize this?

**Comment by RikKoedoot** [comment]
Thats strange. This is a light weight launcher i think

**Comment by hmittal90** [comment]
I have checked in memory tab in settings.It is taking around 190 MB.

**Comment by abdullahibneat** [comment]
Doesn't take too many resources for me, average of 50mb in the last 3 hours: https://photos.app.goo.gl/MLNnIAUObDGDQbuo2

**Comment by Chedi07** [comment]
Facing the same problem. It's consuming 191mb.

**Comment by tsdh** [comment]
96 MB here in the last 3 hours.

**Comment by evertonstrack** [comment]
Same here. Average memory usage of 190 MB. But in several moments 325 MB. Nexus 5X Android Oreo final.

**Comment by abdullahibneat** [comment]
Nexus 5X, Android Oreo, now increased from 50MB to 63MB: https://photos.app.goo.gl/Z3Yq98S5bZUSnBFp1

**Comment by attila123456** [comment]
Can confirm. Average 160 MB, max 280 MB, Nexus 6P becomes significantly slower after a while. Not reproducible on stock Google Now launcher with similar configuration. Too bad because this launcher is a huge improvement otherwise.

**Comment by abdullahibneat** [comment]
Today, average 59MB (max 112MB) on Nexus 5X, Android Oreo: https://photos.app.goo.gl/yIJV66etveELNypk2

**Comment by raf555** [comment]
On my Asus Zenfone 3
Average : 3.9 MB
Max : 123 MB
Is it weird ?? :/

**Comment by raf555** [comment]
https://goo.gl/photos/8kPPhKwViUQxjjudA

**Comment by hmittal90** [comment]
@raf555 The image you have shared is showing **rarely running(3%)** that seems like you are not using this launcher as default launcher.I think it should be always running to check memory usage.

**Comment by samirmz** [comment]
I too observed the same issue. In my Samsung J7 device, this launcher took more than expected memory. After I uninstall this launcher, it released about 200 MB of memory for my ph. Please note, I was using this launcher as default launcher.

**Comment by abdullahibneat** [comment]
I checked again now: in the last 24 hours, average of 61MB (max 122MB): https://photos.app.goo.gl/R5IVuo3rJGACRbOz2

**Comment by cesarferreira** [comment]
`Average 221 MB` (amazing launcher tho!)

**Comment by userNeptune** [comment]
yes, it's a bit poorly optimized, maximum process so far was 272MB

**Comment by abdullahibneat** [comment]
In the past 24hr it used an average of 59MB (max 106MB): https://photos.app.goo.gl/PZuTYopJXxrKHu812

**Comment by ArtemSmaznov** [comment]
Also noted my RAM quite high in usage after installing the launcher and it seems to slow the device a bit. Usually have 180mb ram used by it and sometimes it goes above 300. Don't think I ever had it below 150mb. 

Nexus 6
Android 7.1
https://photos.app.goo.gl/TeRgSZV4D5iqtWtD3

**Comment by jimmykane** [comment]
Same issue here. Even worse. 
![screenshot_20170916-102442](https://user-images.githubusercontent.com/1268723/30510717-834dfb04-9ac9-11e7-90bc-0fc7d5f3babf.png)

**Comment by amirzaidi** [comment]
For some reason your usage tracking bundles the Play Store and Play services too, mine doesn't do that but it includes the Google app too resulting in high memory usage. I don't think the launcher itself uses much memory but the tracking just counts other apps too. 

> it seems to slow the device a bit

Using RAM does not, in any way, slow the device unless you use so much that the system lacks the RAM to normally function. A Nexus 6 with 3GB of RAM should not be affected by slowdowns even IF it were to use 300MB of RAM.

TLDR: It counts the Google apps which are connected to the launcher for the memory usage.

**Comment by jimmykane** [comment]
@amirzaidi Right. I also show that it also counts the google apps. However, sometimes the background is redrawn. All widgets as well stop working for a few and then the memory usage goes down. 
300mb with 2GB on Nexus 5X is not a few btw. For example lunching the camera app and taking some bursts or HDR+ can result to the above behaviour (crashing widgets). 

You can also notice that sometimes the wallpaper goes back to default it the daily ones are selected when the above described behaviour happens.

**Comment by abdullahibneat** [comment]
I confirm the Google App's RAM usage is included in all above readings (on Oreo, go to Settings -> System -> Developer options -> Running services -> Launcher3, scroll down for all external running processes, i.e. Google play services, Google app, Gmal, etc.)

@jimmykane I am of the idea that RAM is there to be used as much as possible instead of leaving it mostly free all the time. I actually never experienced any crashes recently with my normal to slightly-heavy usage (I'm currently working on an AR project and actually test the app on my Nexus 5X). But if you feel like 2GB of RAM is too little for your usage, I recommend you sell your device and buy a better one.
